test/lib/jdk/test/whitebox/WhiteBox.java

```diff
@@ -165,7 +165,52 @@ public         void addToSystemClassLoaderSearch(String segment) {
   }
 
   // G1
+
   public native boolean g1InConcurrentMark();
+  public native int g1CompletedConcurrentMarkCycles();
+
+  // Perform a complete concurrent GC cycle, using concurrent GC breakpoints.
+  // Completes any in-progress cycle before performing the requested cycle.
+  // Returns true if the cycle completed successfully.  If the cycle was not
+  // successful (e.g. it was aborted), then throws RuntimeException if
+  // errorIfFail is true, returning false otherwise.
+  public boolean g1RunConcurrentGC(boolean errorIfFail) {
```

Comment:
Failure detection as described in the comment above is turned off for all test cases. Please verify if that is intentional.

Reply:
Drat!  That's NOT intentional.  Fixing and retesting now.  Thanks for spotting it.